### ISSUE TITLE
AllowForAlignment and Style/ExtraSpacing doesn't allow right-align of = against <<

### ISSUE THREAD

**Issue by DMA57361** [bbatsov/rubocop#2101]
While using the latest master - including the fix for #2098 - I'm getting an offence for an `=` which is right aligned against a `<<`, which itself is then aligned with several other things. I think this is something that should be acceptable. (To be clear, I was getting this before reporting 2098 so that's probably not the cause, but I was going home for the day and didn't have time to construct an example for both yesterday).

Here's a contrived example file that demonstrates:
``` ruby
# Some right-aligning single operators against double-width operators are OK
value  = a_number
value += more_number if want_bigger_number

# Double double-width seems OK as well
array += [1,2,3]
array << another_number if include_another_number

# But when it's the << operator and a single-width we get offenses
a_variable  = [1,2,3]
an_array   << another_number if include_another_number

short             = 'wut'
much_much_longer << :symbol

# When the single-widths aligned with something else, it passes
errors       << some_error if failure
full_message  = errors.join
failures     += 1
```

And the results when run through Rubocop with only this cop enabled:
```
example.rb:10:11: C: Style/ExtraSpacing: Unnecessary spacing detected.
a_variable  = [1,2,3]
          ^
example.rb:11:9: C: Style/ExtraSpacing: Unnecessary spacing detected.
an_array   << another_number if include_another_number
        ^^
example.rb:13:6: C: Style/ExtraSpacing: Unnecessary spacing detected.
short             = 'wut'
     ^^^^^^^^^^^^
example.rb:17:7: C: Style/ExtraSpacing: Unnecessary spacing detected.
errors       << some_error if failure
      ^^^^^^
```
My gut feeling is that all of these should probably be acceptable.

**Comment by jonas054** [comment]
Right hand side alignment is supported only for assignment operators. That's enough IMO. Align other operators on the left hand side.

**Comment by bbatsov** [comment]
I agree with Jonas - that's not an issue, it's like this by design.

**Comment by DMA57361** [comment]
Sorry, having typed this reply I realise I probably should have included this specific code in the original message. My simplified example doesn't really do a very good job of demonstrating what I meant.

---

OK, it just looks odd when in a larger group; you can get a sort of stepping effect to the operators and single-width operators in that block might not all align.

This is the exact code that was getting an offense (for the top line):
``` ruby
last_message_time      = message["created_at"] - customer_waiting_since
all_customer_waits    << last_message_time
total_customer_wait   += last_message_time
max_unreplied_time     = [last_message_time, max_unreplied_time || 0].max
customer_waiting_since = nil
```
that is auto-corrected to the following (losing the alignment in the first line entirely):
``` ruby
last_message_time = message["created_at"] - customer_waiting_since
all_customer_waits    << last_message_time
total_customer_wait   += last_message_time
max_unreplied_time     = [last_message_time, max_unreplied_time || 0].max
customer_waiting_since = nil
```
It would also accept that first `=` being one space earlier, like this (creating a step effect and misaligning the first = with the later ones):
``` ruby
last_message_time     = message["created_at"] - customer_waiting_since
all_customer_waits    << last_message_time
total_customer_wait   += last_message_time
max_unreplied_time     = [last_message_time, max_unreplied_time || 0].max
customer_waiting_since = nil
```
Neither of those two acceptable versions look particularly nice to my eyes, but we've only this one case (in the two projects I've tried master on so far) so I doubt it's a common construct - I'll just add a disable comment when this version comes out.

**Comment by jmehnle** [comment]
The solution is to not use an overly opinionated tool like Rubocop. >:-|
